subprojects/groovy-json/src/main/java/groovy/json/JsonOutput.java

```diff
@@ -601,4 +280,966 @@ public String toString() {
+         *
+         * @param name of the field
+         * @return true if that field is being excluded, else false
+         */
+        boolean isExcludingFieldsNamed(String name) {
+            return hasExcludedFieldNames && excludedFieldNames.contains(name);
+        }
+
+        /**
+         * Package-private helper method used by StreamingJsonBuilder.
+         *
+         * @param value an instance of an object
+         * @return true if values like this are being excluded, else false
+         */
+        boolean isExcludingValues(Object value) {
+            if (value == null) {
+                if (excludeNulls) {
+                    return true;
+                }
+            } else {
+                if (shouldExcludeType(value.getClass())) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+    }
+
+    /**
+     * A converter that handles converting a given type to a JSON value
+     * using a closure.
+     */
+    private static class Converter {
+
+        private final Class<?> type;
+        private final Closure<? extends CharSequence> closure;
+        private final int paramCount;
+
+        static Converter of(Class<?> type, Closure<? extends CharSequence> closure) {
+            return new Converter(type, closure);
+        }
+
+        private Converter(Class<?> type, Closure<? extends CharSequence> closure) {
+            if (type == null) {
+                throw new NullPointerException("Type parameter must not be null");
+            }
+            if (closure == null) {
+                throw new NullPointerException("Closure parameter must not be null");
+            }
+
+            int paramCount = closure.getMaximumNumberOfParameters();
+            if (paramCount < 1) {
+                throw new IllegalArgumentException("Closure must accept at least one parameter");
+            }
+            Class<?> param1 = closure.getParameterTypes()[0];
+            if (!param1.isAssignableFrom(type)) {
+                throw new IllegalArgumentException("Expected first parameter to be of type: " + type.toString());
+            }
+            if (paramCount > 1) {
+                Class<?> param2 = closure.getParameterTypes()[1];
+                if (!param2.isAssignableFrom(String.class)) {
+                    throw new IllegalArgumentException("Expected second parameter to be of type: " + String.class.toString());
+                }
+            }
+            this.type = type;
+            this.closure = closure;
+            this.paramCount = paramCount;
+        }
+
+        /**
+         * Returns {@code true} if this converter can handle conversions
+         * of the given type.
+         *
+         * @param type the type of the object to convert
+         * @return true if this converter can successfully convert values of
+         *      the given type to a JSON value
+         */
+        boolean handles(Class<?> type) {
```

Comment:
I think my rationale here was that since it was a private nested class that is only used internally as a helper that package-private was the appropriate visibility.  But since the override methods are public it may make sense to be consistent and make these public too.